### PR TITLE
Community Announcements

### PR DESCRIPTION
**Pull request by elliotBraem**
_Resolves #540_

This pull request holds the full announcements feature with [contract changes](https://github.com/NEAR-DevHub/neardevhub-contract/pull/79). 

[Testnet Preview](https://test.near.org/devhub-dev.testnet/widget/app?page=community&handle=loud)

**Note**: If developing, use the command:
```cmd
npm run dev:testnet -- -a devhub-dev.testnet -c bodevhub.testnet -n dev
```

# TODO

- [x] Set replacements and track bodevhub.testnet contract
- [x] Attach 2N when creating a community
- [x] Creating community should create dedicated account ([Successful Transaction](https://testnet.nearblocks.io/txns/4t75yejYiN55o7zRxyECQd7n81HbWmMyoazwfJSFCSGk#execution))
- [x] [Communities](https://test.near.social/devhub-dev.testnet/widget/app?page=communities) should show when queried
- [x] Feed module should be adapted to allow an adapter ([Successful set_community_socialdb](https://testnet.nearblocks.io/txns/DQEAXoGCxeCw18Tm89vwZ85b66KQgR1hST7Hw221UNFk), followed by [Successful social.set](https://testnet.nearblocks.io/txns/DQEAXoGCxeCw18Tm89vwZ85b66KQgR1hST7Hw221UNFk#execution). [Post](https://test.near.org/s/p?a=loud.community.bodevhub.testnet&b=152147829) on main feed)
- [x] Preview post should show the post item 
- [x] Commenting (does not work in testnet)
- [x] Test coverage
     - [x] Compose does not show for unauthenticated users
     - [x] Compose shows for admins and moderators users
     - [x] Contract call is correct after commit
     - [x] Comment button is visible
     - [x] Like button is visible
     - [x] A post shows in feed (socialdb query)
     - [x] A comment shows on post in feed (socialdb query)
     - [x] A post shows in feed (near-query-api query)
     - [x] A comment shows on post in feed (near-query-api query)

# Acceptance Criteria

**DevHub community page requirements:**
- [x] There is a new Discussions tab, which is the second tab that users see when they land on a given community
- [x] All users see a share update input text area field (e.g. What's happening)
  - [x] The share update input text area field contains all the same elements from the near.org, including supporting markdown, image update, as well as the preview.
  - [x] Users can type their update, preview it, upload image, and click post. Upon post, the update shows up in the discussions section with the user's profile/name.
- [x] The discussion feed is sorted by latest, but users can also filter by recent comments
- [x] Users can click Load More to load additional posts
- [x] Users can interact with discussions posts (like, add comment, copy link to post, or share).
- [ ] Users can report posts (clicking the three dots). _Note: @ailisp will handle this part in a separate ticket. Please check in with him._

**near.org and near social requirements:**
- [x] Any discussion posted on DevHub communities will also show up on near.org and near social gateways

**Backend storage requirements:**
- [x] Use Social DB and follow near.org conventions-

# Mockups (See discussion section)
https://www.figma.com/file/6bdpB5IrHztN7E40Rjle1N/Community-Feature%3A-Announcements?type=design&node-id=12%3A6010&mode=design&t=M1jwBsC4NvLrn35p-1

# Technical Details
See technical considerations from https://github.com/NEAR-DevHub/neardevhub-bos/issues/540

### CODE REVIEW

**Comment by elliotBraem** [comment]
I realize I still need to complete test coverage, otherwise ready for review

**Comment by ailisp** [comment]
@elliotBraem Great job!

File: src/devhub/components/layout/LikeButton/Faces.jsx [hunk 1]
```diff
@@ -61,16 +61,16 @@ return (
       {renderFaces.map((accountId, i) => (
         <a
           key={i}
-          href={`#/mob.near/widget/ProfilePage?accountId=${accountId}`}
+          href={`#/${REPL_MOB}/widget/ProfilePage?accountId=${accountId}`}
```
Comment (ailisp) [inline src/devhub/components/layout/LikeButton/Faces.jsx:64]: Nice, we can have these testnet widgets now

**Comment by ailisp** [comment]
The code looks good, but I'm unable to see the preview. stucking at this screen:

**Review by ailisp — APPROVED** [review]
This is an awesome work! Thank you

**Pull request merged**